app/src/main/java/nu/yona/app/ui/dashboard/SingleDayActivityDetailFragment.java

```diff
@@ -511,6 +509,23 @@ private void updateDayActivityData(DayActivity dayActivity)
 		viewPager.setCurrentItem(dayActivityList.indexOf(dayActivity));
 		updateFlow(dayActivityList.indexOf(dayActivity));
 		YonaActivity.getActivity().showLoadingView(false, null);
+		setUpYonaHeaderTheme();
+		setDayDetailActivityTitle();
+		setUpUserDetailsInHeader();
+
+	}
+
+	private void setUpYonaHeaderTheme()
+	{
+		if (activity.getYonaGoal().getNickName().equals(YonaApplication.getEventChangeManager().getDataState().getUser().getNickname()))
```

Comment:
This is wrong. Nicknames are not unique, so when the nickname in the message equals the current user's nickname, you know nothing.

To know whether the user is the current one, you need to compare the user URL.

Reply:
```yona:user``` is same when user gets a comment on his day activity or when he gets a reply on buddy activity comment he made. 
Here, intention is to show the difference in theme when the conversation is happening on buddy day activity. 
I believe I can do a string search in day detail URL matching for 'buddies' instead.

Reply:
Do you have a wiki page prescribing this styling difference?

Reply:
https://wiki.yona.nu/display/DEV/Activity+View+-+Day+-+NoGo

Reply:
https://wiki.yona.nu/display/DEV/Activity+View+-+Day+-+Time+Bucket

Reply:
But this is the day details view, right? Not a notification view. These are covered [here:](https://wiki.yona.nu/display/DEV/Me+-+Notifications).

Reply:
Tap on notification will take the user to respective day details view.

Reply:
So you need to find or create a view rather than set a theme on a notification, right?

These views are identical except that the header of that view has a different theme. When rendering that view, you know whether it's for a buddy or for yourself because the linked user differs. Can't you vary the rendering of the activity view based on that?

Reply:
I am not setting a theme on notification. The code above is in SingleDayActivityDetailFragment.java which shows details of day activity of given user (Buddy or self.
You are right that the header (bar color, user icon position and color) is different for both. 
 Linked user you said above is ```yona:user``` in the Notification object right ? It seems to be same in both the cases.

Reply:
You're right. I was wrong with the class name. However, for a day activity detail, you don't need to get the user from the notification but can take it from the day activity detail itself. That has a linked user too and that's what I was referring to.

Reply:
`{
	"spread": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
	"totalActivityDurationMinutes": 0,
	"goalAccomplished": true,
	"totalMinutesBeyondGoal": 0,
	"timeZoneId": "Europe/Amsterdam",
	"date": "2018-10-22",
	"_links": {
		"self": {
			"href": "https://app.prd.yona.nu/users/2099aba8-794d-4df6-a4ed-4fef9660e007/buddies/aa6a673a-4b87-4183-9d04-035366e2afcb/activity/days/2018-10-22/details/77ce88b9-b537-4655-9c0a-b19fcefceb95"
		},
		"yona:goal": {
			"href": "https://app.prd.yona.nu/users/2099aba8-794d-4df6-a4ed-4fef9660e007/buddies/aa6a673a-4b87-4183-9d04-035366e2afcb/goals/77ce88b9-b537-4655-9c0a-b19fcefceb95"
		},
		"yona:messages": {
			"href": "https://app.prd.yona.nu/users/2099aba8-794d-4df6-a4ed-4fef9660e007/buddies/aa6a673a-4b87-4183-9d04-035366e2afcb/activity/days/2018-10-22/details/77ce88b9-b537-4655-9c0a-b19fcefceb95/messages/"
		},
		"yona:addComment": {
			"href": "https://app.prd.yona.nu/users/2099aba8-794d-4df6-a4ed-4fef9660e007/buddies/aa6a673a-4b87-4183-9d04-035366e2afcb/activity/days/2018-10-22/details/77ce88b9-b537-4655-9c0a-b19fcefceb95/messages/"
		},
		"prev": {
			"href": "https://app.prd.yona.nu/users/2099aba8-794d-4df6-a4ed-4fef9660e007/buddies/aa6a673a-4b87-4183-9d04-035366e2afcb/activity/days/2018-10-21/details/77ce88b9-b537-4655-9c0a-b19fcefceb95"
		},
		"next": {
			"href": "https://app.prd.yona.nu/users/2099aba8-794d-4df6-a4ed-4fef9660e007/buddies/aa6a673a-4b87-4183-9d04-035366e2afcb/activity/days/2018-10-23/details/77ce88b9-b537-4655-9c0a-b19fcefceb95"
		},
		"curies": [{
			"href": "http://dev.yona.nu/rels/{rel}",
			"name": "yona",
			"templated": true
		}]
	}
}`

there is no `yona:user` in the day activity details.

Reply:
This evening, I'll look into this. If there is no simple future-proof way to distinguish your own day activities from those of your buddies, I'll create a marker for it.

Reply:
Thanks bert. That would make the fix easy!

Reply:
See yonadev/yona-server#513

Reply:
The PR is merged and the build is deployed to production, so you can use the new link to tell whether it's a buddy activity or an own one.

Reply:
is this available in app.prd.yona.nu ?

Reply:
Yes, it is deployed to production.

Reply:
Ok. I noticed your test case. I could see the ```yona:buddy``` is present in the response.